src/Workspaces/Core/Portable/FindSymbols/SymbolTree/SymbolTreeInfo_Metadata.cs

```diff
@@ -146,11 +147,26 @@ private static Metadata GetMetadataNoThrow(PortableExecutableReference reference
             }
         }
 
+        [PerformanceSensitive("https://github.com/dotnet/roslyn/issues/33131", AllowCaptures = false)]
         public static Checksum GetMetadataChecksum(
             Solution solution, PortableExecutableReference reference, CancellationToken cancellationToken)
         {
             // We can reuse the index for any given reference as long as it hasn't changed.
             // So our checksum is just the checksum for the PEReference itself.
+            // First see if the value is already in the cache, to avoid an allocation if possible.
+            if (ChecksumCache.TryGetValue(reference, out var cached))
+            {
+                return cached;
+            }
+
+            // Break things up to the fast path above and this slow path were we allocate a closure.
+            return GetMetadataChecksumSlow(solution, reference, cancellationToken);
+        }
+
+        [MethodImpl(MethodImplOptions.NoInlining)]
```

Comment:
📝 This is not necessary. Even if the JIT inlines the method, it will not relocate the allocation prior to the fast-path return.

```suggestion
```

Reply:
Good to know.  Thanks!